packages/provider/src/providers/external-window-provider.ts

```diff
@@ -20,7 +20,7 @@ export class ExternalWindowProvider implements AsyncSendable {
   private networkPayload?: NetworkConfig
   private events: EventEmitter<EventType, any> = new EventEmitter()
 
-  constructor(walletAppURL: string) {
+  constructor(walletAppURL: string, chainId?: number) {
```

Comment:
how come we need to pass chainId to the ExternalWindowProvider constructor?

Reply:
looks like you're not using the argument in fact, so probably an early idea and we can remove it now -- which makes more sense since the ExternalWindowProvider would be chainId agnostic, it would be the requests that would be directed at a specific chainId